meilisearch-http/src/helpers/meilisearch.rs

```diff
@@ -526,7 +527,15 @@ fn calculate_highlights(
                 let value: Vec<_> = value.chars().collect();
                 let mut highlighted_value = String::new();
                 let mut index = 0;
-                for m in matches {
+
+                let longest_matches = matches
+                    .iter()
+                    .group_by(|m| m.start)
+                    .into_iter()
+                    .map(|(_key, group)| group.last().unwrap())
+                    .collect::<Vec<_>>();
+
+                for m in longest_matches {
```

Comment:
I think you can avoid collecting this into a vec and directly iterate trought them.

```suggestion
                    .map(|(_key, group)| group.last().unwrap());

                for m in longest_matches {
```